core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -1356,7 +1364,10 @@ private[kafka] class Processor(
   private[network] def channel(connectionId: String): Option[KafkaChannel] =
     Option(selector.channel(connectionId))
 
-  def start(): Unit = thread.start()
+  def start(): Unit = {
+    thread.start()
+    started = true;
```

Comment:
Nit: Scala doesn't require the semicolon
```suggestion
    started = true
```